plugin/trino-hive/src/test/java/io/trino/plugin/hive/ion/IonPageSourceSmokeTest.java

```diff
@@ -114,6 +115,55 @@ public void testProjectedColumn()
                 1);
     }
 
+    @Test
+    public void testPageSourceWithNativeTrinoDisabled()
+            throws IOException
+    {
+        List<HiveColumnHandle> tableColumns = List.of(
+                toHiveBaseColumnHandle("foo", INTEGER, 0),
+                toHiveBaseColumnHandle("bar", VARCHAR, 1));
+        String ionText = "{ foo: 31, bar: baz } { foo: 31, bar: \"baz\" }";
+
+        TrinoFileSystemFactory fileSystemFactory = new MemoryFileSystemFactory();
+        Location location = Location.of("memory:///test.ion");
+
+        // by default Ion native Trino integration is disabled
+        HiveConfig hiveConfig = new HiveConfig();
+
+        final ConnectorSession session = getHiveSession(hiveConfig);
+
+        writeIonTextFile(ionText, location, fileSystemFactory.create(session));
+
+        IonPageSourceFactory factory = new IonPageSourceFactory(fileSystemFactory, hiveConfig);
+
+        long length = fileSystemFactory.create(session).newInputFile(location).length();
+        long nowMillis = Instant.now().toEpochMilli();
+
+        final Map<String, String> tableProperties = ImmutableMap.<String, String>builder()
+                .put(LIST_COLUMNS, tableColumns.stream().map(HiveColumnHandle::getName).collect(Collectors.joining(",")))
+                .put(LIST_COLUMN_TYPES, tableColumns.stream().map(HiveColumnHandle::getHiveType).map(HiveType::toString).collect(Collectors.joining(",")))
+                .buildOrThrow();
```

Comment:
FYI. In the [PR #4](https://github.com/rmarrowstone/trino/pull/4/commits/2d4b4514d14b4197a8a154b2afc58793f1d8993e): there's method to get the tableProperties. Feel free to merge this PR if get approved. I can make the update in [PR #4](https://github.com/rmarrowstone/trino/pull/4/commits/2d4b4514d14b4197a8a154b2afc58793f1d8993e).